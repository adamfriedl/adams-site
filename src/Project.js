import React from 'react'
import { Heading, Text, Flex} from 'rebass'

const Project = ({
  title = 'My Title',
  description = 'My description',
  github = '#',
  demo = '#'
}) =>

  <div>

    <Heading 
      f={[2, 3]}
      mt={[2, 3]}
      children={title}
    />

    <Text
      my={[1, 2]}
      children={description}
    />

    <Flex
      mb={[2, 3]}
    >
      <Text
        is='a'
        href={github}
        children='Github'
        mr={2}   
      />

      <Text
        is='a'
        href={demo}
        children='Demo'            
      />
    </Flex>

  </div>

  export default Project