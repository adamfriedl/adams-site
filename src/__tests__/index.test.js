import React from 'react';
import ReactDOM from 'react-dom';
import { create as render } from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Project from '../Project';
import Footer from '../Footer';


it('should render a project', () => {
  const wrapper = render(<Project />)
  expect(wrapper).toMatchSnapshot() 
})

test('Project works when rendered shallow', () => {
  const wrapper = shallow(<Project />)
  expect(wrapper).toMatchSnapshot()
})