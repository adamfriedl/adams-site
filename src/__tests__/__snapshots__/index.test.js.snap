// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project works when rendered shallow 1`] = `
<div>
  <Styled(Styled(Base))
    f={
      Array [
        2,
        3,
      ]
    }
    mt={
      Array [
        2,
        3,
      ]
    }
  >
    My Title
  </Styled(Styled(Base))>
  <Styled(Styled(Base))
    my={
      Array [
        1,
        2,
      ]
    }
  >
    My description
  </Styled(Styled(Base))>
  <Styled(Styled(styled.div))
    mb={
      Array [
        2,
        3,
      ]
    }
  >
    <Styled(Styled(Base))
      href="#"
      mr={2}
    >
      Github
    </Styled(Styled(Base))>
    <Styled(Styled(Base))
      href="#"
    >
      Demo
    </Styled(Styled(Base))>
  </Styled(Styled(styled.div))>
</div>
`;

exports[`should render a project 1`] = `
<div>
  <h2
    className="sc-dxgOiQ jydelQ sc-kpOJdX igVeNG sc-kGXeez liylew sc-kgoBCf hEvGne"
    is={null}
  >
    My Title
  </h2>
  <p
    className="sc-kGXeez liylew sc-kgoBCf cimYuh"
    is={null}
  >
    My description
  </p>
  <div
    className="sc-bxivhb bUZGRr sc-htpNat eLQHN sc-bdVaJa lctIqv"
  >
    <a
      className="sc-fjdhpX nsimD sc-jTzLTM fMyqly"
      href="#"
      is={null}
    >
      Github
    </a>
    <a
      className="sc-fjdhpX nsimD sc-jTzLTM iLwSIb"
      href="#"
      is={null}
    >
      Demo
    </a>
  </div>
</div>
`;
